larreco/QuadVtx/EvalVtx_module.cc

```diff
@@ -0,0 +1,135 @@
+// Chris Backhouse - c.backhouse@ucl.ac.uk - Oct 2019
+
+#include "fhiclcpp/ParameterSet.h"
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Core/EDAnalyzer.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/Handle.h"
+#include "art/Framework/Services/Registry/ServiceHandle.h"
+#include "art_root_io/TFileService.h"
+#include "canvas/Persistency/Common/FindManyP.h"
+
+#include "lardataobj/RecoBase/Vertex.h"
+#include "lardataobj/RecoBase/PFParticle.h"
+#include "nusimdata/SimulationBase/MCTruth.h"
+
+#include "TTree.h"
+
+namespace{
+  int gEvt;
+  double gTrueX, gTrueY, gTrueZ;
+  std::vector<double> gRecoX, gRecoY, gRecoZ;
+}
+
+namespace quad
+{
+
+class EvalVtx: public art::EDAnalyzer
+{
+public:
+  explicit EvalVtx(const fhicl::ParameterSet& pset);
+
+  void analyze(const art::Event& evt) override;
+
+protected:
```

Comment:
Doesn't that mean that protected vs private is completely moot? I generally always use protected because I think you kind of have to trust classes inheriting from you, but I won't fight too hard if protected->private is a requirement.

Reply:
Sorry, Chris, I didn't give you enough explanation.  A few issues:

- The C++ core guidelines discourage protected data for various reasons (https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-protected).  A real-life example where protected data is problematic is the *messagefacility*.   That system has been incredibly difficult to maintain due to its use of protected data: derived classes mess around with base-class state.  We're slowly cleaning it up, but it's difficult to do so.  It's better to use a default protection of `private`, and then to expose data members when it's actually required.
- Even though nothing will (or *should*) inherit from a module, specifying `private` instead of `protected` does not hurt anything, is fewer characters to type, and it keeps with the pattern recommended by the core guidelines.  People don't have to think about why one module uses `protected` data and the other `private`.
- Specifying `protected` in a module could imply to someone down the road that the module was designed to be inherited from.  That's not something we want people to do.  In fact, I sometimes type `class MyModule final : public art::EDProducer {...}`, with the `final` keyword to be even more explicit.

Reply:
OK, done